arccnet/data_generation/timeseries/sdo_processing.py

```diff
@@ -51,7 +50,33 @@
 ]
 
 
-def read_data(hek_path: str, srs_path: str, size: int, duration: int):
+def rand_select(table, size, types: list):
+    r"""
+    Randomly selects targets from provided table and list of data subsets.
+
+    Parameters
+    ----------
+        table : `Astropy.Table`
+            Provided full table of records.
+        size : `int`
+            The size of the returned subsets.
+        types: `list`
+            List of datatypes to include.
+
+    Returns
+    -------
+        comb_sample : `Astropy.Table`
+            Returned random subset containing size records from each data subset.
+    """
+    selection = []
+    for type in types:
+        subtable = table[table["category"] == type]
```

Comment:
Since the input just an astroy table this columns name should be parameter also need some tests.